scripts/extensions/videoEditor/src/VideoTimeline/VideoTimeline.tsx

```diff
@@ -0,0 +1,251 @@
+import * as React from 'react';
+import {debounce, isEmpty} from 'lodash';
+import {BarIcon} from './BarIcon';
+import {ListThumbnails} from './ListThumbnails';
+import VideoEditorContext from '../VideoEditorContext';
+import {IThumbnail} from '../interfaces';
+
+interface IProps {
+    video: React.RefObject<HTMLVideoElement>;
+    thumbnails: Array<IThumbnail>;
+    trim: {
+        start: number;
+        end: number;
+    };
+    onTrim: (start: number, end: number, runCheckIsDirty: boolean) => void;
+}
+interface IState {
+    currentTime: number;
+    thumnailsRender: Array<IThumbnail>;
+}
+
+export class VideoTimeline extends React.Component<IProps, IState> {
+    static contextType = VideoEditorContext;
+    private wrapper: React.RefObject<HTMLDivElement>;
+    private controlbar: React.RefObject<HTMLDivElement>;
+    private intervalTimer: number;
+    private PositionX: any;
+
+    constructor(props: IProps) {
+        super(props);
+        this.state = {
+            currentTime: 0,
+            thumnailsRender: [],
+        };
+        this.wrapper = React.createRef();
+        this.controlbar = React.createRef();
+        this.intervalTimer = 0;
+    }
+
+    componentDidMount() {
+        // call tick every 100ms to update current time state
```

Comment:
We've used `timeupdate` event. However, the loop time of it is 250ms, it's little bit long, we cannot setting it, it base on system, [references](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/timeupdate_event). So we have a issue, It make Video is like [delay](https://drive.google.com/open?id=1a1fxqz3QHwpU0Zdj167rUJlTW_rbcpMe).
Should we keep the interval or use `timeupdate`?

Reply:
Do you mean that the timeline progress indication is delayed a bit as the video is playing? I could see that too. Keep the interval solution then. Add a comment with explanation which tick is used instead of `timeupdate`